File: javascript/hamming/hamming.js
var Hamming = function() {}

Hamming.prototype.compute = function(strandA, strandB){
  var strand = strandA.split('');

  if (strand.length != strandB.length) {
    throw 'DNA strands must be of equal length.'
  }

  return strand.reduce((total, letter, index) => {
    return (strand[index] == strandB[index]) ? total : ++total;
  }, 0)
}

module.exports = Hamming;

File: java/hamming/src/main/java/Hamming.java
public class Hamming {
    public static int compute(String strandA, String strandB) {
        if (strandA.length() != strandB.length()) {
            throw new IllegalArgumentException("Strands must be equal length");
        }

        int totalDifferences = 0;

        for (int i = 0; i < strandA.length(); i++) {
          if (strandA.charAt(i) != strandB.charAt(i)) { totalDifferences += 1; }
        }

        return totalDifferences;
    }
}